src/main/java/org/springframework/data/redis/core/RedisTemplate.java

```diff
@@ -715,11 +715,14 @@ public Long getExpire(K key, final TimeUnit timeUnit) {
 		return execute(new RedisCallback<Long>() {
 
 			public Long doInRedis(RedisConnection connection) {
```

Comment:
`getExpire(K, TimeUnit)` has a bug if it's used with transactions/pipelining. It throws in the first place a `NullPointerException`. If the call is guarded against `null` access the results return the original `ttl`/`pTtl` value and not the converted one. I think we should move the conversion into the connections as that's the place where we're able to perform a conversion. Adding `pTtl(K, TimeUnit)` and `ttl(K, TimeUnit)` to `RedisKeyCommands` should fix that bug.